Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -299,10 +299,14 @@ private extension UInt8 {
         case .unknown:
             for header in self.headers {
                 if self.buffer.equalCaseInsensitiveASCII(view: "connection".utf8, at: header.name) {
-                    if self.buffer.equalCaseInsensitiveASCII(view: "close".utf8, at: header.value) {
-                        return false
-                    }
-                    return self.buffer.equalCaseInsensitiveASCII(view: "keep-alive".utf8, at: header.value)
+                    let states = self.buffer.extractCaseInsensitiveConstantStrings(
+                        ["keep-alive", "close"],
```

Comment:
I think this API is overkill. 

The generally useful construct is, given an entire header block, receive a maximally decomposed list header (that is, split on all commas, aware of quoted strings, including repeated header field names). Ideally we’d have a lazy iterator that can produce `ByteBuffer` slices of this form. 

That construct is useful because it solves a more general problem, rather that a specific “look for this token” model.